docs/product-user-guide/src/main/asciidoc/rules-attributes-ref.adoc

```diff
@@ -0,0 +1,60 @@
+
+|`no-loop`
+|A Boolean value. When the option is selected, the rule cannot be reactivated (looped) if a consequence of the rule re-triggers a previously met condition. When the condition is not selected, the rule can be looped in these circumstances.
+
+|`agenda-group`
+|A string identifying an agenda group to which you want to assign the rule. Agenda groups allow you to partition the agenda to provide more execution control over groups of rules. Only rules in the agenda group that have acquired a focus are able to be activated. *<@Michael (4 of 6): Can you tell me what a focus is and what it's used for? It comes up again in `auto-focus`.>*
+
+|`activation-group`
+|A string identifying an activation (or XOR) group to which you want to assign the rule. In activation groups, only one rule can be activated. The first rule to fire will cancel all pending activations of all rules in the activation group.
+
+|`duration`
+|A long integer value defining the duration of time in minutes during which the rule can be activated. After the specified duration of time, the rule cannot be activated. *<@Michael (5 of 6): Is this accurate? And is it for certain in minutes? I see this as basically an expiration in minutes on the rule (similar to `date-expires`, but in minutes, not a date) and that it can fire within the duration period but not after. BUT the description in the GDT UI in Central says basically the opposite: "The duration dictates that the rule will fire after a specified duration, if it is still true." i.e., that it canNOT (right?) fire during the duration but can after, like a pause button. That's how I read it. Do you know which it is?>*
+
+|`timer`
+|A string identifying either `int` (interval) or `cron` timer definition for scheduling the rule.
+
+|`calendars`
+|A Quartz calendar definition for scheduling the rule.
+
+|`auto-focus`
+|A Boolean value, applicable only to rules within agenda groups. When the option is selected, the next time the rule is activated, a focus is automatically given to the agenda group to which the rule is assigned.
+
+|`lock-on-active`
+|A Boolean value, applicable only to rules within ruleflow groups or agenda groups. When the option is selected, the next time the ruleflow group for the rule becomes active or the agenda group for the rule receives a focus, the rule cannot be activated again until the ruleflow group is no longer active or the agenda group loses the focus. This is a stronger version of the `no-loop` attribute, because the activation of a matching rule is discarded regardless of the origin of the update (not only by the rule itself). This attribute is ideal for calculation rules where you have a number of rules that modify a fact and you do not want any rule re-matching and firing again.
+
+|`ruleflow-group`
+|A string identifying a ruleflow group. In ruleflow groups, rules can fire only when the group is active. *<@Michael (6 of 6): Do you know how ruleflow groups become active?>*
```

Comment:
@sterobin Nope sorry! @krisv @mswiderski should be able to advise.

Reply:
@sterobin Ruleflow groups are activated when using business rule tasks inside a process (sometimes also called a RuleFlow).  A business rule task defines the name of the ruleflow group that will be activated.  Once the process reaches the node (after starting the ruleflow), the ruleflow group with the specified name will be activated.

Reply:
Thanks @krisv ! So, does this even apply to Decision Manager (Drools) or just to Business Automation (jBPM)? Since DM/Drools doesn't support business processes and tasks, from my understanding.

@manstis, are there any other rule attributes that you can think of that apply only to Business Automation (jBPM) and therefore wouldn't be suitable in a DM/Drools doc?

Reply:
@sterobin Yes, as Michael mentioned some basic process capabilities (called RuleFlow) are included in RHDM as well, which includes the business rule task.